src/authentication/authentication-hapi-plugin.ts

```diff
@@ -1,23 +1,34 @@
 import * as Boom from 'boom';
 import * as auth from 'hapi-auth-jwt2';
 import { inject, injectable } from 'inversify';
-import * as qs from 'querystring';
+import * as Joi from 'joi';
+import * as Knex from 'knex';
 
 import * as Hapi from '../server/hapi';
 import { HapiPlugin } from '../server/hapi-register';
-import { Group, User } from '../shared/gitlab';
+import { Group } from '../shared/gitlab';
 import { GitlabClient } from '../shared/gitlab-client';
+import { Logger, loggerInjectSymbol } from '../shared/logger';
+import { adminTeamNameInjectSymbol, charlesKnexInjectSymbol } from '../shared/types';
+import { generateTeamToken, TeamToken, teamTokenQuery, validateTeamToken } from './team-token';
 import { AccessToken, jwtOptionsInjectSymbol } from './types';
 
+const randomstring = require('randomstring');
```

Comment:
Since the API for this module seems to be rather small, we could consider adding type definitions for it.